UrlsSpiders/BBCUrlsSpider.py

```diff
@@ -0,0 +1,90 @@
+import scrapy
+import os
+import re
+import csv
+import datetime
+import operator
+from string import printable
+from scrapy.spiders import CrawlSpider, Rule
+from scrapy.linkextractors import LinkExtractor
+from scrapy.linkextractors.lxmlhtml import LxmlLinkExtractor
+from scrapy.signalmanager import SignalManager
+from scrapy.exceptions import CloseSpider
+from scrapy import signals
+from pydispatch import dispatcher
+from distinctUrls import distinct
+
+'''
+	The '".\urls-results\2019-BBC-visitedUrls.txt"' file was not generated 
+	using this spider but by filtring the results from the BBC spider for 
+	the jargon project.
+'''
+
+class WebSite(scrapy.Item):
+	url = scrapy.Field()
+	year = scrapy.Field()
+
+class BBCUrlsSpider(CrawlSpider):
+	name = "BBCUrlsSpider"
+	allowed_domains = ["bbc.com"]
+
+	YEAR = '2019'
+	LIMIT_URLS_COUNT = 1000
+	MATCH = r'.*\bwww\.bbc\.com\/news\/science-environment-.*$'
+
+	visitedUrls = []
+	start_urls  = ['https://www.bbc.com/news/science_and_environment']
+	
+	def __init__(self, *args, **kwargs):
+		super(BBCUrlsSpider, self).__init__(*args, **kwargs)
+		SignalManager(dispatcher.Any).connect(self.spiderClosed, signals.spider_closed)
+
+	rules = (	
+		Rule(LinkExtractor(allow=(r'(\bnews\b)\D+(-\d+)$'),
+						    deny=(r'.*(m\.|\.test\.|\.stage\.|%|comments|\/live\/|\/athlete\/|\/weather\/).*'),
+							restrict_xpaths="//body"),
+			 				callback='parseNews',
+			 				follow=True),
+    )
+				
+	def parseAll(self, response, year):
+		item = WebSite()
+		item['url'] = response.url		
+		item['year'] = str(year)
+
+		if item['year'] == self.YEAR and re.match(self.MATCH, item['url']) != None:
```

Comment:
why not use the implemented rules for this?

Reply:
Because the implemented rules is used for the crawling too, it defines which urls will be followed and how to parse them.

Reply:
you can have rule for parse, and rule for crawl and parse with the callback parameter